modules/libcom/src/osi/os/posix/osdSock.c

```diff
@@ -99,6 +99,22 @@ LIBCOM_API SOCKET epicsStdCall epicsSocketCreate (
             close ( sock );
             sock = INVALID_SOCKET;
         }
+
+#ifdef __linux__
+#  ifndef IP_MULTICAST_ALL
+#    define IP_MULTICAST_ALL		49
```

Comment:
Is it wise to define and use this macro if the C library headers don't actually define it? It has probably been included in glibc for a long time and I'm fine with doing this if it's available, but we don't test on other libc implementations and they might not define it. My first instinct would be that if it isn't defined, we shouldn't try to use it at all.

Reply:
Wisdom is a tricky thing.  My first instinct was the same as yours.  However, I found at least one environment (python manylinux1) still has old Linux 2.6 era headers.  As far as I can tell, IP socket option 49 was unused prior to the introduction of `IP_MULTICAST_ALL`.  Since this is all wrapped by `#ifdef __linux__` it should effect other OSs.  So anyone still running such an old kernel (circa RHEL5) should only see this new warning.